docs/remote/containers-advanced.md

```diff
@@ -72,26 +72,89 @@ You can use the `workspaceMount` property in `devcontainer.json` to change the a
+"workspaceFolder": "/workspace"
+"runArgs": [
+    "-v", "try-node-node_modules:/workspace/node_modules"
+]
+```
+
+The `workspaceMount` property ensures the source code is always mounted into the container in the same location. The `runArgs` command then creates a named volume called `try-node-node_modules` and mounts the `node_modules` sub-folder into it instead. The rest of the files then come from the local filesystem.
+
+Since this sample [runs VS Code as non-root user](#adding-a-non-root-user-to-your-dev-container) in the container called "node", a `postCreateCommand` is included to make sure the user can access the `node_modules` folder.
+
+```json
+"workspaceMount": "src=${localWorkspaceFolder},dst=/workspace,type=bind,consistency=cached",
+"workspaceFolder": "/workspace",
+"runArgs": [
+    "-u", "node",
+    "-v", "try-node-node_modules:/workspacee/node_modules"
+],
+"postCreateCommand": "sudo chown node:node node_modules"
+```
+
+If you've already built the container and connected to it, run **Remote-Containers: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up the change. Otherwise run **Remote-Containers: Open Folder in Container...** to connect to the container.
+
+**Docker Compose**:
+In this case, you will add the named volume to your Docker Compose file (or an [extended one](/docs/remote/containers.md#extending-your-docker-compose-file-for-development)) for the appropriate service(s). For example:
+
+```yaml
+version: '3'
+services:
+  your-service-name-here:
+    volumes:
+       # Or wherever you've mounted your source code
+      - .:/workspace
+      - your-service-name-here-node_modules: /workspace/node_modules
+
+volumes:
+  - try-node-node_modules:
+```
+
+You'll also want to be sure the `workspaceFolder` property in `devcontainer.json` matches the place your actual source code mounted:
+
+```json
+"workspaceFolder": "/workspace"
+```
+
+If you're running in the container with a [user other than root](#adding-a-non-root-user-to-your-dev-container), add a postCreateCommand to update the owner of the folder you mount since it may have been mounted as root.
+
+```json
+"workspaceFolder": "/workspace",
+"postCreateCommand": "sudo chown your-user-name-here node_modules"
+```
+
+If you've already built the container and connected to it, run **Remote-Containers: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up the change. Otherwise run **Remote-Containers: Open Folder in Container...** to connect to the container.
+
+### Update the mount consistency to delegated in Docker for Mac
+
+By default, the Remote - Containers extension uses the [cached mount consistency](https://docs.docker.com/docker-for-mac/osxfs-caching/) on macOS since this provides a good mix between performance and write guarantees on the host OS. However, you can opt to use the `delegated` consistency instead if you do not expect to be writing to the same file in both locations very often. Just follow these steps:
```

Comment:
There is also a user setting that is applied when `"workspaceMount"` does not override the default mount:
![image](https://user-images.githubusercontent.com/9205389/63831171-53bd2100-c96e-11e9-9132-e86eece088cd.png)

Settings ID: `"remote.containers.workspaceMountConsistency"`.